### PR TITLE
Changing files structure, adding styled-components

### PR DESCRIPTION
**Pull request by kucabix**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -90,42 +68,40 @@ class App extends React.Component {
 
   render() {
     const {fromCurrency, toCurrency, exchangeRate, fromCountry, toCountry, isLoading, selectedCurrency} = this.state
-    const loadingClass = isLoading ? '-loading' : ''
-    const selectClass = selectedCurrency.length ? '' : '-none'
+    const isSelected = selectedCurrency.length ? true : false
```
Comment (tomeroth) [inline src/App.js]: oneliner ifs are nice but in this case if u want boolean values anyways, better approach would be
`!!selectedCurrency.length`, get some info about double exclamation mark in js

File: src/components/CurrencyInfo.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+const CurrencyDiv = styled.div`
+font-family: Montserrat;
+display: flex;
+flex-direction: column;
+margin: 10px auto;
+flex-wrap: wrap;
+width: 70vw;
+background-color: #eee;
+border-radius: 10px;
+padding: 20px;
```
Comment (tomeroth) [inline src/components/CurrencyInfo.js]: some tabs in code formating here would be nice

File: src/elements/Button.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+export default styled.button`
+  font-family: Montserrat;
+  font-weight: bold;
+  line-height: 38px;
+  border: 1px solid #888;
+  border-radius: 5px;
+  background-color: #50a6e0;
+  padding: 0 20px;
+  &:hover{
+    background-color: #4793c6;
```
Comment (tomeroth) [inline src/elements/Button.js]: this is a .js file, why so many static css here? Created coresponing class in css files for this btn

Comment (tomeroth) [reply src/elements/Button.js]: same in other files like this one

File: src/js/get_data.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+const axios = require('axios')
+
+const getExchangeRate = async (fromCurrency, toCurrency) => {
+  const response = await axios.get('http://data.fixer.io/api/latest?access_key=9b82443638e1ca1466eb99939d777ee8')
```
Comment (tomeroth) [inline src/js/get_data.js]: good practice would be receiving link as an attribute to this func, give this link when calling it. In this way you are not closed for calling this func with other url in the future etc

File: src/js/get_data.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+  const rate = response.data.rates
+  const base = 1 / rate[fromCurrency]
+  const exchangeRate = base * rate[toCurrency]
+  if (isNaN(exchangeRate)) {
+    throw new Error (`Unable to get currencies ${fromCurrency} and ${toCurrency}`)
```
Comment (tomeroth) [inline src/js/get_data.js]: too late to check `isNan`? Shouldn't you check whether `rate[toCurrency]` is Nan instead?

File: src/js/get_data.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  }
+  return `Exchange rate: ${exchangeRate.toFixed(3)}`
+}
+
+const getCountries = async (toCurrency) => {
+  try {
+    const response = await axios.get(`https://restcountries.eu/rest/v2/currency/${toCurrency}`)
+    return response.data.map(country => country.name)
```
Comment (tomeroth) [inline src/js/get_data.js]: this try has to much going on inside if u want to catch only `get` response then do data mapping after and outside of `try/catch`. Just a suggestion

File: src/js/get_data.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+  } catch (e) {
+    throw new Error (`Unable to get countries that use ${toCurrency}`)
+  }
+}
+
+const getData = async (currencies) => {
+  const fromCountry = await getCountries(currencies[0])
+  const toCountry = await getCountries(currencies[1])
+  const exchange = await getExchangeRate(currencies[0], currencies[1])
+  return {fromCountry: fromCountry, toCountry: toCountry, exchange: exchange}
```
Comment (tomeroth) [inline src/js/get_data.js]: not sure but i think u could just go with `{fromCountry, toCountry, exchange}`

File: src/App.js (hunk 1 above)
Comment (kucabix) [reply src/App.js]: Solved

File: src/elements/Button.js (hunk 3 above)
Comment (kucabix) [reply src/elements/Button.js]: I've left only files with styled-components, used in the main api, to make it more readable.

File: src/js/get_data.js (hunk 5 above)
Comment (kucabix) [reply src/js/get_data.js]: By checking exchangeRate, I'm checking if both rate[toCurrency] and rate[fromCurrency] were fetched properly. If not, exchangeRate will be NaN.

File: src/js/get_data.js (hunk 7 above)
Comment (kucabix) [reply src/js/get_data.js]: true

File: src/js/get_data.js (hunk 4 above)
Comment (kucabix) [reply src/js/get_data.js]: solved

File: src/js/get_data.js (hunk 5 above)
Comment (tomeroth) [reply src/js/get_data.js]: y, welcome to js world, you can multiply `undefined` and check at the end, but that doesn`t seem like a good practice, u still don't know which one failed, `base` or `exchangeRate`, maybe `rate`, maybe `fromCurrency` etc. The question is do you even care in this case. Leave it as it is, but consider other approaches in the future

**Pull request merged**